MyArrayList/src/main/java/org/example/MyArrayList.java

```diff
@@ -0,0 +1,310 @@
+package org.example;
+
+import java.util.*;
+
+public class MyArrayList<E> implements Collection<E> {
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private Object[] elementData;
+    private int size;
+
+    public MyArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+    }
+
+    public MyArrayList(int capacity) {
+        if (capacity >= 0) {
+            this.elementData = new Object[capacity];
+        } else {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    public MyArrayList(Collection<? extends E> c) {
+        this(c.size());
+        Object[] collectionToArray = c.toArray();
+        size = collectionToArray.length;
+        System.arraycopy(collectionToArray, 0, elementData, 0, size);
+    }
+
+    public void bubbleSort(Comparator<? super E> c) {
```

Comment:
This variant sorting less productive than quick sort. Nick said about bubble sort: it isn't allowance.